pallets/stream-payment/src/lib.rs

```diff
@@ -738,6 +777,11 @@ pub mod pallet {
                 .ok_or(Error::<T>::StreamIdOverflow)?;
             NextStreamId::<T>::set(next_stream_id);
 
+            // TODO save the holded amount to storage
+            let hold_amount = T::OpenStreamHoldAmount::get();
+            T::Currency::hold(&HoldReason::StreamOpened.into(), &origin, hold_amount)?;
```

Comment:
@nanocryk do we envision cases in which we dont apply this hold? (Maybe if the stream-opening comes from data-preservers, which already will do a hold per se

Reply:
I've changed the `open_stream_returns_id` function to accept an `opening_deposit` https://github.com/moondance-labs/tanssi/pull/493/files#diff-6a7c7b4db06d7b82cddd52c77c149e7ee2e502bdc12e7291b9f3743349ec377fR763.
When it's called by the extrinsic this is set to the value in the runtime, and it can also be called by a future trait to be used by other pallets to set their own deposit if needed.
Would this suffice?

Reply:
yes, thanks!